apps/full-site-editing/full-site-editing-plugin/premium-content/premium-content.php

```diff
@@ -118,8 +137,61 @@ function premium_content_block_init() {
 			'style'           => 'premium-content-frontend',
 		)
 	);
+}
+
+/**
+ * Load static assets for the editor.
+ */
+function load_static_editor() {
+	$info         = get_script_info();
+	$script_asset = $info['script_asset'];
+	$url_path     = $info['url_path'];
+	$dir          = $info['dir'];
+
+	$index_js = 'dist/premium-content.js';
+	wp_register_script(
+		'premium-content-editor',
+		"$url_path/$index_js",
+		$script_asset['dependencies'],
+		$script_asset['version'],
+		false
```

Comment:
These parts have a bit of cognitive overhead and not needed abstraction. How would you feel making them more simple like 
https://github.com/Automattic/wp-calypso/blob/2b83937b085691ba66d162c35ca4a28ff48a5ce4/apps/full-site-editing/full-site-editing-plugin/newspack-blocks/index.php#L54-L60
Ultimately, not using `get_script_info` too ?

I know that of course they are not part of this PR and we can think of it in a consecutive PR if needed.

Reply:
Sure.  I took a number of mechanical steps to simplify a bit.  Step through them or check out the final results in the files changed tab.  I was able to get $url_path and $dir down to defined constants.  `get_script_info()` turned into `get_script_asset()`, which is a bit simpler, but the helper function still exists.

c79a066950 - PremiumContent: Extract $info['url_path'], $info['dir'] to defined constants
707e407101 - PremiumContent: get_script_info() -> get_script_asset().
bd1b3b4da2 - Premium content: Inline $url_path variable
698e918af2 -  Premium Content: Inline $dir

I think this is better, let me know if you have a strategy for getting rid of `get_script_asset()`.

Reply:
Thanks @mreishus for addressing my points, much appreciated! Indeed code seems better now!
I took the liberty to create a PR https://github.com/Automattic/wp-calypso/pull/45052 suggesting how we may get rid of `get_script_asset()` as also some other minors that got in my way ;-). Please feel free to amend or merge. I am approving current PR 🚀 !

Also, tested your changes and everything seems well nothing has broken ;-)!